python/lsst/dax/webserv/client/httpclient.py

```diff
@@ -0,0 +1,121 @@
+# LSST Data Management System
+# Copyright 2015 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+
+"""
+Low level client for HTTP interaction.
+@author  Brian Van Klaveren, SLAC
+"""
+
+import requests
+from urllib import quote as _qt
+
+class BaseHttpClient:
+    """Request/Response Helper class for all HTTP endpoint clients"""
+
+    def __init__(self, url, auth_strategy=None, *args, **kwargs):
+        self.base_url = url
+        self.auth_strategy = auth_strategy
+
+    def _target(self, endpoint, path):
+        def _resolve(_path, part):
+            """Sanitize all the slashes!"""
+            _path = _path.rstrip("/")
+            part = part.lstrip("/")
+            return "/".join((_path, part)) if len(part) else _path
+        newBaseUrl = _resolve(self.base_url, endpoint)
+        return _resolve(newBaseUrl, path)
+
+    def doRequest(self, httpMethod, endpoint, target, params=None, data=None, **kwargs):
+        """If necessary, target should have been quoted"""
+        target = self._target(endpoint, target)
+        headers = kwargs["headers"] if "headers" in kwargs else None
+        requests_method = getattr(requests, httpMethod)
```

Comment:
Is httpMethod is expected to be lowercase? This should be documented, usually method name is uppercase. Or force-convert it to lower case here.